File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int cur = 0;
        List<Integer> list = new ArrayList<>();
        
        while(progresses[progresses.length - 1] != -1) {
            for(int i = 0; i < progresses.length; i++) {
                progresses[i] += speeds[i];
            }
            
            int count = 0;
            
            while(cur < progresses.length && progresses[cur] >= 100) {
                count++;
                progresses[cur] = -1;
                cur++;
            }
            
            if(count != 0) {
                list.add(count);   
            }
        }
        
        int[] ans = new int[list.size()];
        for(int i = 0; i < list.size(); i++) {
            ans[i] = list.get(i);
        }
        
        return ans;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
import math

def solution(progresses, speeds):
    p_dq = deque(progresses)
    s_dq = deque(speeds)
    answer = []
    
    count = 0
    day = 0
    for p, s in zip(p_dq, s_dq):
        if day == 0:
            day = math.ceil((100 - p) / s)
            count += 1
        elif p + (s * day) >= 100:
            count += 1
        else:
            answer.append(count)
            count = 1
            day = math.ceil((100 - p) / s)
            
    if count > 0:
        answer.append(count)
    
    return answer